### PR TITLE
chore: make doc extractor node also can extract text by file extension

### PR DESCRIPTION
**Pull request by hjlarry**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [ ] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

currently, extract doc by the mimetype is not so much reliable. for example,  the markdown file will always raise error:

image

so extract doc by the filename when can't recognize the mimetype can improve success rate

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

test locally

- [ ] Test A
- [ ] Test B

### CODE REVIEW

**Comment by laipz8200** [comment]
Do you mind providing a file that will cause the error? According to my attempt, the .md file can be correctly identified as text/markdown.

**Comment by hjlarry** [comment]
> Do you mind providing a file that will cause the error? According to my attempt, the .md file can be correctly identified as text/markdown.

I tried these files, the mimetype is always `application/octet-stream` :
[2024-09-27.md](https://github.com/user-attachments/files/17483716/2024-09-27.md)
[Markdown 101 (Example).md](https://github.com/user-attachments/files/17483719/Markdown.101.Example.md)
[Just chat.zip](https://github.com/user-attachments/files/17483743/Just.chat.zip)

I think this behavior is depends on the browser,  I use `Edge` of windows11 for test

**Comment by hjlarry** [comment]
someone else  encounter this issue https://github.com/langgenius/dify/issues/9757

I think for the remote_file extract by mimetype, for the local file extract by extension  is more reasonable

**Comment by yaoice** [comment]
lgtm

File: api/core/workflow/nodes/document_extractor/node.py [hunk 1]
```diff
@@ -107,6 +107,33 @@ def _extract_text(*, file_content: bytes, mime_type: str) -> str:
         raise UnsupportedFileTypeError(f"Unsupported MIME type: {mime_type}")
 
 
+def _extract_text_by_file_extension(*, file_content: bytes, file_extension: str) -> str:
+    """Extract text from a file based on its file extension."""
+    match file_extension:
```
Comment (laipz8200) [inline api/core/workflow/nodes/document_extractor/node.py:112]: Have you tested based on the latest code? The `File.extension` should include the dot.

Comment (hjlarry) [reply api/core/workflow/nodes/document_extractor/node.py:112]: just merge the latest branch and add dot for it

**Review by laipz8200 — APPROVED** [review]
LGTM

**Pull request merged**